instrumentation/netty/src/test/java/brave/netty/HttpSnoopyServerInitializer.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2012 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package brave.netty;
+
+import brave.http.HttpTracing;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.handler.codec.http.HttpObjectAggregator;
+import io.netty.handler.codec.http.HttpRequestDecoder;
+import io.netty.handler.codec.http.HttpResponseEncoder;
+
+public class HttpSnoopyServerInitializer extends ChannelInitializer<SocketChannel> {
```

Comment:
Netty uses EmbeddedChannel for junit test, but only for ChannelHandler(in or out).
Sending http request into EmbeddedChannel,after other ChannelHandler processed(the class you want to test),get the result(http response or processed http request) out from EmbeddedChannel.
So there is no other server class can accept http request from other client(Maby I didn't find it).